src/Model/AggregatesTrait.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atk4\Data\Model;
+
+use Atk4\Data\Model;
+
+/**
+ * Provides aggregation methods.
+ */
+trait AggregatesTrait
+{
+    /**
+     * @param array|object $seed
+     *
+     * @return Aggregate
+     *
+     * @see Aggregate::withAggregateField.
+     */
+    public function withAggregateField(string $name, $seed = []): Model
+    {
+        return (new Aggregate($this))->withAggregateField($name, $seed);
+    }
+
+    /**
+     * @return Aggregate
+     *
+     * @see Aggregate::groupBy.
+     */
+    public function groupBy(array $fields, array $aggregateExpressions = []): Model
```

Comment:
function name is brutally confusing as it returns a new Model, it should be named like `createAggregate`

Reply:
we should probably rename to `setGroupBy` to be consistent with Model methods naming

Reply:
fixed